src/jaxoplanet/orbits/keplerian.py

```diff
@@ -169,240 +308,90 @@ class Body(eqx.Module):
-            self.cos_asc_node = cos_asc_node
+            self.sin_asc_node = body.sin_asc_node
+            self.cos_asc_node = body.cos_asc_node
 
         # Handle eccentric and circular orbits
-        self.eccentricity = eccentricity
-        if eccentricity is None:
-            if sin_omega_peri is not None:
-                raise ValueError("Cannot specify omega_peri without eccentricity")
-
-            M0 = jnpu.full_like(self.period, 0.5 * jnp.pi)
+        self.eccentricity = body.eccentricity
+        if self.eccentricity is None:
+            M0 = jnpu.full_like(self.period, 0.5 * jnp.pi)  # type: ignore
             incl_factor = 1
         else:
-            if self.sin_omega_peri is None:
-                raise ValueError("Must specify omega_peri for eccentric orbits")
-
+            assert self.sin_omega_peri is not None
+            assert self.cos_omega_peri is not None
             opsw = 1 + self.sin_omega_peri
             E0 = 2 * jnpu.arctan2(
-                jnpu.sqrt(1 - eccentricity) * self.cos_omega_peri,
-                jnpu.sqrt(1 + eccentricity) * opsw,
+                jnpu.sqrt(1 - self.eccentricity) * self.cos_omega_peri,
+                jnpu.sqrt(1 + self.eccentricity) * opsw,
             )
-            M0 = E0 - eccentricity * jnpu.sin(E0)
+            M0 = E0 - self.eccentricity * jnpu.sin(E0)
 
-            ome2 = 1 - eccentricity**2
-            incl_factor = (1 + eccentricity * self.sin_omega_peri) / ome2
+            ome2 = 1 - self.eccentricity**2
+            incl_factor = (1 + self.eccentricity * self.sin_omega_peri) / ome2
 
         # Handle inclined orbits
         dcosidb = incl_factor * central.radius / self.semimajor
-        if impact_param is not None:
-            if inclination is not None:
-                raise ValueError("Cannot specify both inclination and impact_param")
-            self.impact_param = impact_param
-            self.cos_inclination = dcosidb * impact_param
+        if body.impact_param is not None:
+            self.impact_param = body.impact_param
+            self.cos_inclination = dcosidb * body.impact_param
             self.sin_inclination = jnpu.sqrt(1 - self.cos_inclination**2)
-        elif inclination is not None:
-            self.cos_inclination = jnpu.cos(inclination)
-            self.sin_inclination = jnpu.sin(inclination)
+        elif body.inclination is not None:
+            self.cos_inclination = jnpu.cos(body.inclination)
+            self.sin_inclination = jnpu.sin(body.inclination)
             self.impact_param = self.cos_inclination / dcosidb
         else:
-            self.impact_param = jnpu.zeros_like(self.period)
-            self.cos_inclination = jnpu.zeros_like(self.period)
-            self.sin_inclination = jnpu.ones_like(self.period)
+            z = jnp.zeros_like(self.period.magnitude) * ureg.dimensionless
+            self.impact_param = z
+            self.cos_inclination = z
+            self.sin_inclination = z
```

Comment:
@dfm: I think `self.sin_inclination` is supposed to be initialized with `1`s instead of `0`s (`z`)!

Reply:
Nice catch @soichiro-hattori! I wasn't sure you could so I just pushed the changes. I hope it's fine!

Reply:
Good catch @soichiro-hattori!! Thanks for tracking that down!